modules/core/src/main/scala/qq/droste/algebras.scala

```diff
@@ -39,94 +39,89 @@ final class GAlgebra[S, F[_], A](val run: F[S] => A) extends AnyVal {
         Gather.zip(gather, that.gather))
   }
 }
 
-final class GCoalgebra[S, F[_], A](val run: A => F[S]) extends AnyVal {
+final class GCoalgebra[F[_], A, S](val run: A => F[S]) extends AnyVal {
   def apply(a: A): F[S] =
     run(a)
 
-  def scatter(scatter: Scatter[S, F, A]): GCoalgebra.Scattered[S, F, A] =
+  def scatter(scatter: Scatter[F, A, S]): GCoalgebra.Scattered[F, A, S] =
     GCoalgebra.Scattered(this, scatter)
 
   def lift[M[_]](implicit M: Applicative[M]): A => M[F[S]] =
     a => run(a).pure[M]
 
-  def compose[Z](f: GCoalgebra[A, F, Z])(implicit F: FlatMap[F]): GCoalgebra[S, F, Z] =
+  def compose[Z](f: GCoalgebra[F, Z, A])(implicit F: FlatMap[F]): GCoalgebra[F, Z, S] =
     GCoalgebra(z => f(z) flatMap run)
 
-  def andThen[T](f: GCoalgebra[T, F, S])(implicit F: FlatMap[F]): GCoalgebra[T, F, A] =
+  def andThen[T](f: GCoalgebra[F, S, T])(implicit F: FlatMap[F]): GCoalgebra[F, A, T] =
     f compose this
 
   def toKleisli: Kleisli[F, A, S] =
     Kleisli(run)
 }
 
 object GCoalgebra extends GCoalgebraInstances {
-  def apply[S, F[_], A](run: A => F[S]): GCoalgebra[S, F, A] =
+  def apply[F[_], A, S](run: A => F[S]): GCoalgebra[F, A, S] =
     new GCoalgebra(run)
 
-  final case class Scattered[S, F[_], A](
-    coalgebra: GCoalgebra[S, F, A],
-    scatter: Scatter[S, F, A]
+  final case class Scattered[F[_], A, S](
+    coalgebra: GCoalgebra[F, A, S],
+    scatter: Scatter[F, A, S]
   )
 }
 
 // instances
 
 private[droste] sealed trait GAlgebraInstances {
 
-  implicit def drosteArrowForGAlgebra[F[_]: Comonad]: Arrow[GAlgebra[?, F, ?]] =
+  implicit def drosteArrowForGAlgebra[F[_]: Comonad]: Arrow[GAlgebra[F, ?, ?]] =
     new GAlgebraArrow
 }
 
-private[droste] class GAlgebraArrow[F[_]: Comonad] extends Arrow[GAlgebra[?, F, ?]] {
-  def lift[A, B](f: A => B): GAlgebra[A, F, B] =
+private[droste] class GAlgebraArrow[F[_]: Comonad] extends Arrow[GAlgebra[F, ?, ?]] {
+  def lift[A, B](f: A => B): GAlgebra[F, A, B] =
     GAlgebra(fa => f(fa.extract))
-  def compose[A, B, C](f: GAlgebra[B, F, C], g: GAlgebra[A, F, B]): GAlgebra[A, F, C] =
+  def compose[A, B, C](f: GAlgebra[F, B, C], g: GAlgebra[F, A, B]): GAlgebra[F, A, C] =
     f compose g
-  override def andThen[A, B, C](f: GAlgebra[A, F, B], g: GAlgebra[B, F, C]): GAlgebra[A, F, C] =
+  override def andThen[A, B, C](f: GAlgebra[F, A, B], g: GAlgebra[F, B, C]): GAlgebra[F, A, C] =
     f andThen g
-  def first[A, B, C](f: GAlgebra[A, F, B]): GAlgebra[(A, C), F, (B, C)] =
+  def first[A, B, C](f: GAlgebra[F, A, B]): GAlgebra[F, (A, C), (B, C)] =
     GAlgebra(fac => (f.run(fac.map(_._1)), fac.map(_._2).extract))
 }
 
 private[droste] sealed trait GCoalgebraInstances {
-  // Note: It's very difficult to summon this instance since it's
-  // right-to-left instead of left-to-right. So...
-  // TODO: Re-evaluate order of type parameters for algebras
-  implicit def drosteArrowForGCoalgebra[F[_]: Monad]: Arrow[λ[(α, β) => GCoalgebra[β, F, α]]] =
```

Comment:
This "issue" instigated this change. But I also think this arrangement is a bit more readable. Gather/Scatter aren't left-to-right and instead just mirror the parameter ordering of Algebra/Coalgebra, respectively.